### PR TITLE
chore(release): v2.34.0 with codebase-memory-mcp 0.9.0

### PR DESCRIPTION
**Pull request by Lyellr88**
Bumps the wrapped code-index binary from 0.8.1 to 0.9.0 and makes the wrapper compatible with it.

0.9.0 paginates tools/list, returning 8 of 14 tools with a nextCursor. CbmClient.list_tools() issued a single request, so check_schema() saw the remaining 6 as removed upstream and refused to start the code graph. The client now follows the cursor to the end. Verified against both real binaries: 0.8.1 answers unpaginated and stays a single request, 0.9.0 pages and yields all 14.

The loop terminates on nextCursor being absent rather than falsy, because a cursor is opaque per MCP and 0 or "" are valid values a truthiness check would silently treat as the end. A repeated cursor raises rather than looping. Both are covered by tests and were confirmed load-bearing by reverting each one.

PINNED_CBM_VERSION moves to 0.9.0. Its comment recorded that 0.8.1's binary misreported itself as 0.10.0; 0.9.0 reports correctly, so the note now reads as history with a warning not to assume it holds next bump.

Also:

- test(graph): drain the console index job to a terminal state instead of abandoning it after the 202. The endpoint starts a daemon thread that outlived its test and indexed a tmp_path pytest had already deleted, surfacing as a child-process error under an unrelated later test. Adds coverage for the worker's own single-flight lock release, which had none.
- chore(dev): scripts/typecheck.py runs mypy on core/ against an error baseline, failing only when the count rises. Local only for now.
- pyproject: enable the pydantic mypy plugin and scope follow_imports, which removes two false "missing required argument" errors and stops mypy reporting 227 errors across 55 files while checking 26.
- CONTRIBUTORS: credit Muneeb Ahmad for the RRF bake-off reference implementation and the #113 consolidation report.

Refs #91, #113.

## Summary by CodeRabbit

* **New Features**
  * Improved code-graph startup reliability with complete paginated operation discovery.
  * Added safeguards against repeated pagination cursors and startup version mismatches.
  * Added fallback handling for code-indexing startup issues.

* **Bug Fixes**
  * Improved cleanup and lock handling when indexing jobs fail.

* **Documentation**
  * Updated installation guides, Docker metadata, and project documentation for version 2.34.0.
  * Added the latest changelog entry and contributor attribution.

* **Tests**
  * Added coverage for pagination, cursor edge cases, version handling, and indexing failure recovery.

### CODE REVIEW

File: docs/INSTALL-LINUX.md [hunk 1]
```diff
@@ -320,7 +320,7 @@ curl -s http://localhost:8001/health
 {
   "status": "healthy",
   "service": "MARM MCP Server",
-  "version": "2.33.1",
+  "version": "2.34.0",
```
[the remark this answers is not in the document]
Comment (Lyellr88) [reply docs/INSTALL-LINUX.md:323]: false claim
